Sources/Apodini/Semantic Model Builder/GraphQL/GraphQLInterfaceExporter.swift

```diff
@@ -0,0 +1,150 @@
+                                                  headers: { 'Content-Type': 'application/json' },
+                                                  body: JSON.stringify(graphQLParams),
+                                                })
+                                                  .then(response => response.json())
+                                                  .catch(() => response.text());
+                                              ReactDOM.render(
+                                                React.createElement(GraphiQL, { fetcher: graphQLFetcher }),
+                                                document.getElementById('graphiql'),
+                                              );
+                                            </script>
+                                          </body>
+                                        </html>
+                                        """
+
+        return Vapor.Response(status: .ok, headers: ["Content-Type": "text/html"], body: html)
+    }
+
+    private func graphqlServer(_ req: Vapor.Request) throws -> EventLoopFuture<String> {
+        guard let body = req.body.string else {
+            throw NetworkError.noBody
+        }
+
+        // Create Swift dict
+        var query = "{}"
+        let data = body.data(using: .utf8)!
+        do {
+            if let jsonArray = try JSONSerialization.jsonObject(with: data, options: .allowFragments) as? [String: AnyObject] {
+                if let q = jsonArray["query"] as? String {
+                    query = q
+                } else {
+                    print("jsonArray doesn't have query parameter")
+                }
+
+            } else {
+                print("bad json")
+                throw NetworkError.noBody
+            }
+        } catch let error as NSError {
+            print(error)
+            throw NetworkError.noBody
+        }
+
+        return try graphql(schema: self.schema,
+                request: query,
+                context: req,
+                eventLoopGroup: req.eventLoop).map { result -> String in
+            result.description
+        }
+
+    }
+
+    // We should have parameter for the result struct values and paramters
+    func export<H: Handler>(_ endpoint: Endpoint<H>) {
+        self.graphQLPath.append(for: endpoint, with: endpoint.createConnectionContext(for: self))
+    }
+
+    func exportParameter<Type: Codable>(_ parameter: EndpointParameter<Type>) -> String {
+        parameter.name
+    }
+
+    func finishedExporting(_ webService: WebServiceModel) {
+        self.schema = self.graphQLPath.generate()
+    }
+
+    func retrieveParameter<Type: Decodable>(_ parameter: EndpointParameter<Type>, for request: GraphQLRequest) throws -> Type?? {
+        if let queryArgDict = request.args.dictionary,
+           let val = queryArgDict[parameter.name] {
+
+            if (val.isNumber) {
+                return val.int as? Type
```

Comment:
Ok now I see what you meant in the work meeting. 
Here's where we'll have to do some work to support: `UInt`, as well as `Enums` and `InputObjects`